actors/verifreg/src/lib.rs

```diff
@@ -762,6 +811,133 @@ where
     )
 }
 
+// Deserializes and validates a token receiver hook payload.
+fn validate_tokens_received(
+    raw: &RawBytes,
+    my_id: u64,
+) -> Result<TokensReceivedParams, ActorError> {
+    let payload: TokensReceivedParams = deserialize(raw, "receiver hook payload")?;
+    // Payload to address must match receiving actor.
+    if payload.to != my_id {
+        return Err(actor_error!(
+            illegal_argument,
+            "token receiver expected to {}, was {}",
+            my_id,
+            payload.to
+        ));
+    }
+    Ok(payload)
+}
+
+// Deserializes and validates a serialized allocation.
+fn validate_alloc_req(
+    serialized: &RawBytes,
+    policy: &Policy,
+    curr_epoch: ChainEpoch,
+    datacap_amount: &DataCap,
+) -> Result<AllocationRequests, ActorError> {
+    let reqs: AllocationRequests = deserialize(serialized, "allocation requests")?;
+
+    let mut allocation_total = StoragePower::zero();
```

Comment:
Why not DataCap?  I think I'm missing something fundamental about how you use these types.  If you use storage power here we should use it in claim_allocations too

Reply:
Ok changed, but I think we should delete the type now.